docs/source/contribution.rst

```diff
@@ -443,3 +287,111 @@ Open ``index.html`` with the browser and see if it is rendered as expected.
+  8 weeks    X.1.0*      Y.0.0b1    --
+ 12 weeks    X.2.0*      Y.0.0rc1   --
+ 16 weeks    --          Y.0.0      Z.0.0a1
+========== =========== =========== ============
+
+(* These might be revision releases)
+
+The dates shown in the left-most column are relative to the release of ``X.0.0rc1``.
+In particular, each revision/minor release is made around four weeks after the previous one of the same major version, and the pre-release of the upcoming major version is made at the same time.
+Whether these releases are revision or minor is determined based on the contents of each update.
+
+Note that there are only three stable releases for the versions ``X.x.x`` in the example table above.
+The number of stable releases may vary depending the development status of the following in this case ``Y`` and its number of required beta versions (a ``b`` followed by a number).
+During the parallel development of ``Y.0.0`` and ``Z.0.0a1``, the version ``Y`` is treated as an **almost-stable version** and ``Z`` is treated as a development version.
+
+If there is a critical bug found in ``X.x.x`` after stopping the development of version ``X``, we may release a hot-fix for this version at any time.
+
+A `milestone for each upcoming release is published on GitHub <https://github.com/chainer/chainer/milestones>`_.
+The GitHub milestones are used to group issues and PRs belonging to a release.
+
+.. _contrib-git-branches:
+
+Git Branches
+~~~~~~~~~~~~
+
+The ``master`` branch is used to develop pre-release versions.
+It means that **alpha, beta, and RC updates are developed at the** ``master`` **branch**.
+This branch contains the most up-to-date source tree that includes features newly added after the latest major version.
+
+The stable version is developed on the ``vN`` branch where "N" reflects the version number (*versioned branch*).
+For example, v3.0.0, v3.1.0, and v3.2.0 are developed on the ``v3`` branch.
+
+A PR from a contributor should in general be targeting the ``master`` branch.
+If the change can and should be applied to the stable version in addition, a member from the core team will make sure it is backported to be included in the next revision update.
+
+If the change is only applicable to the stable version and not to the ``master`` branch, please send it to the versioned branch.
+We basically only accept changes to the latest versioned branch (where the stable version is developed) unless the fix is critical.
+
+If you want to introduce a new feature in the ``master`` branch to the current stable version, please send a *backport PR* to the stable version (the latest ``vN`` branch).
+See the next section for details.
+
+*Note: a change that can be applied to both branches should be sent to the* ``master`` *branch.*
+*Each release of the stable version is also merged with the development version so that changes are reflected in the next major version.*
+
+Feature Backport Pull Requests
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
Shoud we mention the backport tool?

Reply:
Thanks for the tip. I added a sentence dedicated to it.